engine/src/main/resources/swagger/cromwell.yaml

```diff
@@ -716,6 +756,208 @@ definitions:
         type: string
         description: The status of the workflow
         example: Submitted
+  WorkflowDescription:
+    required:
+      - valid
+    properties:
+      valid:
+        type: boolean
+        description: Whether the workflow and inputs as submitted passes validation
+      errors:
+        type: array
+        items:
+          type: string
+        example:
+          - "The 'errors' field will be filled if 'valid' is false"
+          - "We might also provide warnings to a 'valid' workflow here"
+          - "Otherwise, 'errors' will be empty or unspecified"
+        description: The set of validation failure messages
+      name:
+        type: string
+        description: The name of the workflow in the description file
+        example: my_wf
+      inputs:
+        description: A list of inputs for this tool
+        type: array
+        items:
+          $ref: '#/definitions/ToolParameter'
+        example:
+          - name: my_wf.string_input
+            valueType:
+              typeName: String
+            optional: false
+          - name: my_wf.array_input
+            valueType:
+              typeName: Array
+              arrayType:
+                typeName: String
+            optional: false
+          - name: my_wf.optional_input
+            valueType:
+              typeName: Optional
+              optionalType:
+                typeName: String
+            optional: true
+            default: "hello"
+          - name: my_wf.map_input
+            valueType:
+              typeName: Map
+              mapType:
+                keyType:
+                  typeName: String
+                valueType:
+                  typeName: Int
+            optional: false
+          - name: my_wf.object_input
+            valueType:
+              typeName: Object
+              objectFieldTypes:
+                - fieldName: "int_field"
+                  fieldType:
+                    typeName: Int
+                - fieldName: "int_array_field"
+                  fieldType:
+                    typeName: Array
+                    arrayType:
+                      typeName: Int
+            optional: false
+          - name: my_wf.int_string_pair_input
+            valueType:
+              typeName: Tuple
+              tupleTypes:
+                - typeName: Int
```

Comment:
Is it that the order in which `typeName` is provided differentiates between the left and right components of the tuple?